crypto/crypto/src/ed25519.rs

```diff
@@ -117,6 +117,39 @@ impl Ed25519PublicKey {
             Err(_) => Err(CryptoMaterialError::DeserializationError),
         }
     }
+
+    /// Deserialize an Ed25519PublicKey from its representation as an x25519
+    /// public key, along with an indication of sign. This is meant to
+    /// compensate for the poor key storage capabilities of key management
+    /// solutions, and NOT to promote double usage of keys under several
+    /// schemes, which would lead to BAD vulnerabilities.
+    /// Arguments:
+    /// - `x25519_bytes`: bit representation of a public key in clamped
+    ///            Montgomery form, a.k.a. the x25519 public key format.
+    /// - `negative`: whether to interpret the given point as a negative point,
+    ///               as the Montgomery form erases the sign byte. By XEdDSA
+    ///               convention, if you expect to ever convert this back to an
+    ///               x25519 public key, you should pass `false` for this
+    ///               argument.
+    pub fn from_x25519_public_bytes(
```

Comment:
Why do we need x25519 -> ed25519? For a complete picture?
But since these are "dangerous" APIs, may be we should not expose conversion in this direction since we don't need it?

Reply:
For unit tests IIUC. I'll gate it.